src/makeCalculator.js

```diff
@@ -37,7 +37,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    operate: function(cb, value) {
+      this.result = cb(this.result, value);
+
+      return this;
+    },
+    reset: function() {
+      this.result = 0;
+
+      return this;
+    },
+    add: function(result, value) {
+      return result + value;
+    },
+    subtract: function(result, value) {
+      return result - value;
+    },
+    multiply: function(result, value) {
+      return result * value;
+    },
+    divide: function(result, value) {
+      return result / value;
```

Comment:
There should be a check for division by zero in the 'divide' function. In the current implementation, if 'value' is 0, it will return Infinity, which might break further calculations.